### ISSUE TITLE
Support for helm-semantic-or-menu ?

### ISSUE THREAD

**Issue by rogergl** [ananthakumaran/tide#5]
It would be nice if tide would support helm-semantic-or-menu to display a list of current functions and classes in the buffer.

**Comment by dmh43** [comment]
Doesn't Tide already support this?

**Comment by MarcusRiemer** [comment]
It's sort-of supported: All identifiers seem to appear, but they are all labeled as `Function` and have minor annoyances. If having multiple classes in a single file for example, the class properties are not prefixed by the class name which (possibly) leads to loads of identically named entries.

![Everything is a Function](https://cloud.githubusercontent.com/assets/2138773/16663898/3dca99d6-447e-11e6-960c-2be39daef3b7.png)

**Comment by dmh43** [comment]
@MarcusRiemer looks like that functionality is defined in `tide-imenu-index`. Inside this method, we could prefix the name of the method with the name of the class it is defined on.

I think in order to get the items to appear as `class` or `variable` rather than `Function` we'd need to generate the imenu index using a regex. Not sure about this last point though; would have to do more research.

**Comment by NStal** [comment]
@MarcusRiemer Your jsdoc style comment looks cool, how do you generate them?

**Comment by josteink** [comment]
> I think in order to get the items to appear as class or variable rather than Function we'd need to generate the imenu index using a regex. Not sure about this last point though; would have to do more research.

I don't think we specifically need to do that, but from my experience, I would say `imenu` has some quirks...

The following is what I can tell from my experimentation when rewriting imenu-indexing completely from scratch in [csharp-mode](https://github.com/josteink/csharp-mode). I don't guarantee any canonical or eternal correctness to these statements. These are just what I seem to have observed.

Basically if you want something to show as a "class", "interface" or whatever, you have to create a new sub-menu node for it (and add all nested members there). Everything added to the "root" index will show as functions... No exceptions, regexp or not.

When using `helm-imenu` having lots of nested members is probably not a big problem (since helm flattens the list anyway), but for people using the menus... It might be clicky? I don't know.
